worlds/LauncherComponents.py

```diff
@@ -31,9 +32,8 @@ def __init__(self, display_name: str, script_name: Optional[str] = None, frozen_
         self.cli = cli
         self.type = component_type or \
             None if not display_name else \
```

Comment:
while you are at it,
- i don't think `bool(display_name)` should ever be `False`, so we could remove the `None if ...` and change the type hint for `type` to not be optional? if we want hide a button, it'd be better to use a special type or an additional flag rather than having `display_name = ""`